mlx/fast.h

```diff
@@ -39,4 +39,26 @@ array scaled_dot_product_attention(
     const std::optional<array>& mask = std::nullopt,
     StreamOrDevice s = {});
 
+std::tuple<array, array, array> affine_quantize(
+    const array& w,
+    int group_size = 64,
+    int bits = 4,
+    StreamOrDevice s = {});
+
+array affine_quantize_with_params(
+    const array& w,
+    const array& scales,
+    const array& biases,
+    int group_size = 64,
+    int bits = 4,
+    StreamOrDevice s = {});
```

Comment:
Could you explain what the purpose of that one is?

Reply:
Yes it's useful for something like GPTQ where you calculate the quantization scales and biases for a group and then apply them to individual columns of the weight matrix which get updated as you iterate through them (e.g. [here](https://github.com/IST-DASLab/gptq/blob/2d65066eeb06a5c9ff5184d8cebdf33662c67faf/gptq.py#L130)).

I'm not sure if this is the best naming for it (or maybe it should be an overload of `affine_quantize`?).

Reply:
> I'm not sure if this is the best naming for it (or maybe it should be an overload of affine_quantize?).

Indeed `with_params` is a bit vague. Maybe an overload is better.